patternlab/styleguide/source/assets/js/modules/mainNavHamburger.js

```diff
@@ -291,3 +372,177 @@ window.onresize = function () {
+    closeSubMenu();
+  } else {// Close
+    closeNarrowUtilContent();
+    utilNarrowContent.style.pointerEvents = "none";
+  }
+});
+
+function closeNarrowUtilContent() {
+  // utilNarrowContent.classList.add("is-closed");
+  utilNarrowContent.setAttribute("aria-hidden", "true");
+
+  // Button state
+  utilNarrowButton.setAttribute("aria-expanded", "false");
+  utilNarrowButton.setAttribute("aria-pressed", "false");
+
+  utilNarrowContent.style.opacity = "0";
+  utilNarrowContent.style.height = "0";
+
+  /** Slide up. */
+  setTimeout(function timeoutFunction() {
+    utilNarrowContent.style.opacity = "0";
+    utilNarrowContent.style.height = "0";
+  }, 700);
+  setTimeout(function timeoutFunction() {
+    // utilNarrowContent.style.opacity = "0";
+    // utilNarrowContent.style.height = "0";
+
+    utilNarrowContent.classList.add("is-closed");
+  }, 500);
+}
+
+function closeSubMenu() {
+  setTimeout(function timeoutFunction() {
+    let openSubMenu = document.querySelector(".submenu-open");
+    let openSubMenuButton = openSubMenu.querySelector(".js-main-nav-hamburger__top-link");
+    let openSubMenuContent = openSubMenu.querySelector(".js-main-nav-hamburger-content");
+    let openSubMenuContainer = openSubMenu.querySelector(".js-main-nav-hamburger__container");
+
+    //  button
+    openSubMenuButton.setAttribute("aria-expanded", "false");
+
+    // sub menu container div
+    openSubMenuContent.classList.add("is-closed");
+    openSubMenuContent.style.height = "0";
+
+    // sub menu list container
+    openSubMenuContainer.style.opacity = 0;
+
+    // sub menu container list item
+    openSubMenu.classList.remove("submenu-open");
+  }, 500);
+}
+
+// Keyboard navigation
+document.addEventListener("keydown", function (e) {
+  // ESC to close menus.
+  if (e.key === "Escape" || e.which === "27") {
+
+    // Check which menu is open.
+    if (utilNavWide.querySelector(".js-util-nav-content").style.opacity === "1") {// Log in to... in Utility nav bar
+      closeUtilWideContent();
+      utilNavWide.querySelector(".js-util-nav-toggle").focus();
+    }
+    else if (utilNarrowContent.style.opacity === "1") {// Log in to... in Hamburger menu
+      closeNarrowUtilContent();
+      setTimeout(function timeoutFunction() {
+        utilNarrowButton.focus();
+      }, 90);
+    }
+    else {// Menu button
+      closeMenu();
+      setTimeout(function timeoutFunction() {
+        document.querySelector(".js-header-menu-button").focus();
+      }, 100);
+
+      document.querySelector(".js-utility-nav--wide .ma__utility-nav__item  .goog-te-menu-value").removeAttribute("tabindex");
+      document.querySelector(".js-utility-nav--wide .ma__utility-nav__item  .direct-link").removeAttribute("tabindex");
+      document.querySelector(".js-utility-nav--wide .ma__utility-nav__item  .js-util-nav-toggle").removeAttribute("tabindex");
+    }
+
+    // if (body.classList.contains("show-menu") && utilNarrowContent.style.opacity === "0" ) {// Hamburger menu
```

Comment:
Let's remove these comments if no longer needed

Reply:
Keep this for now for the future setting up keyboard navigation work.